src/util/mathematical_expr.h

```diff
@@ -186,6 +186,15 @@ inline void validate_expr(const factorial_power_exprt &value)
   validate_operands(value, 2, "Factorial power must have two operands");
 }
 
+class tuple_exprt : public multi_ary_exprt
```

Comment:
Would it be worth moving this to `std_expr.h` and use it in those places where we currently do use `ID_arguments`?

Reply:
Alternatively I would suggest putting all the `.*_exprt` classes in a new module (separate from `util`) and have one class per header file, but that's a much bigger change.

Reply:
I like this idea, but this is a very intrusive change (a lot of code in `util` already depends on `std_expr.h`:
```
$ git grep 'include.*std_expr.h' src/util/ | wc -l
43
```

Reply:
I think that the arguments/parameters of a C/C++ function are a very different thing from the arguments/parameters of a mathematical function. In particular, there is no tuple building going on in C/C++ or Java.

Reply:
Ok.

Reply:
@romainbrenguier getting slightly off-topic, we could separate the 'irep/expression' related things from the 'non irep utilities', say making a directory named "irep". There could be dummy headers to a transition period.

Reply:
that sounds good